test/common/PtrUnion.cpp

```diff
@@ -0,0 +1,354 @@
+#include "PtrUnion.hpp"
+
+namespace RcclUnitTesting
+{
+  size_t DataTypeToBytes(ncclDataType_t const dataType)
+  {
+    switch (dataType)
+    {
+    case ncclInt8:   return 1;
+    case ncclUint8:  return 1;
+    case ncclInt32:  return 4;
+    case ncclUint32: return 4;
+    case ncclInt64:  return 8;
+    case ncclUint64: return 8;
+    case ncclFloat16: return 2;
+    case ncclFloat32: return 4;
+    case ncclFloat64: return 8;
+    case ncclBfloat16: return 2;
+    default:
+      ERROR("Unsupported datatype (%d)\n", dataType);
+      exit(0);
+    }
+  }
+
+  ErrCode PtrUnion::Attach(void *ptr)
+  {
+    this->ptr = ptr;
+    return TEST_SUCCESS;
+  }
+
+  ErrCode PtrUnion::Attach(PtrUnion ptrUnion)
+  {
+    this->ptr = ptrUnion.ptr;
+    return TEST_SUCCESS;
+  }
+
+  ErrCode PtrUnion::AllocateGpuMem(size_t const numBytes, bool const useManagedMem)
+  {
+    if (numBytes)
+    {
+      if (useManagedMem)
+      {
+        if (hipMallocManaged(&I1, numBytes) != hipSuccess)
+        {
+          ERROR("Unable to allocate managed memory of GPU memory (%lu bytes)\n", numBytes);
+          return TEST_FAIL;
+        }
+      }
+      else
+      {
+        if (hipMalloc(&I1, numBytes) != hipSuccess)
```

Comment:
ccould replace if statement here with `CHECK_HIP(hipMalloc(&I1, numBytes))` for consistency through the tests

Reply:
Unfortunately, this wouldn't provide information about the number of byes